### DISCUSSION TITLE
Contact profile Picture (Url)

### DISCUSSION THREAD

**Discussion by vqmacedo** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#26] [General]
How can I get the Contact profile Picture? (Url) ?

**Comment by gabrieldwight** [comment]
Hi @vqmacedo, I am unsure if the contact profile picture URL is exposed when receiving the message due to privacy. When receiving the message, you can inspect the webhook object if the contacts object contains the URL property.

If sending a contact message the Url will represent the URL information saved for that contact which can be a website for the contact or any additional data.

**Reply by vqmacedo** [reply]
Did you see or implement this call?

https://developers.facebook.com/docs/graph-api/reference/profile-picture-source/

**Reply by gabrieldwight** [reply]
I will check on it to see if the endpoint works with WhatsApp cloud api. I have not yet implemented on the cloud api library

**Reply by gabrieldwight** [reply]
The profile picture source works with Graph API that manages Facebook Pages that support New Page Experience, an update for migrating from pages API. It can't work with Cloud API since one of the endpoints for the profile picture source requires a **page-id** parameter linked to a Facebook page.
